url4aac7e6f55_simon0191_tetris3d_tgz-pJ8-Tetris3dJ8/src/vsdk/toolkit/gui/CameraControllerAquynza.java

```diff
@@ -0,0 +1,438 @@
+//===========================================================================
+
+package vsdk.toolkit.gui;
+
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseWheelEvent;
+//import java.awt.event.KeyEvent; // Do not include, deprecated
+
+import vsdk.toolkit.common.linealAlgebra.Vector3D;
+import vsdk.toolkit.common.linealAlgebra.Matrix4x4;
+import vsdk.toolkit.environment.Camera;
+
+public class CameraControllerAquynza extends CameraController {
+
+    private Camera camera;
+    private int oldMouseX;
+    private int oldMouseY;
+    private double deltaMov;
+
+    public CameraControllerAquynza(Camera camera) {
+        this.camera = camera;
+        oldMouseX = 0;
+        oldMouseY = 0;
+        deltaMov = 0.25;
+    }
+
+    public double getDeltaMovement()
+    {
+        return deltaMov;
+    }
+
+    public void setDeltaMovement(double val)
+    {
+        deltaMov = val;
+    }
+
+    public boolean processMouseEventAwt(MouseEvent mouseEvent) {
+      return true;
+    }
+
+    private double augmentLogarithmic(double val, double EPSILON)
+    {
+        if ( val < 0.001 ) val += 0.0001;
+        else if ( val < 0.01 ) val += 0.001;
+        else if ( val < 0.1 - EPSILON ) val += 0.01;
+        else if ( val < 1 - EPSILON ) val += 0.1;
+        else if ( val < 10 - EPSILON ) val += 1;
+        else if ( val < 100 - EPSILON ) val += 10;
+        else if ( val < 1000 - EPSILON ) val += 100;
+        else if ( val < 10000 - EPSILON ) val += 1000;
+        else if ( val < 100000 - EPSILON ) val += 10000;
+        else if ( val < 1000000 - EPSILON ) val += 100000;
+        else if ( val < 10000000 - EPSILON ) val *= 2;
+        else val = 10000000;
+  
+        return val;
+    }
+  
+    private double diminishLogarithmic(double val, double EPSILON)
+    {
+        if ( val > 10000000 + EPSILON ) val /= 2;
+        else if ( val > 1000000 + EPSILON ) val -= 1000000;
+        else if ( val > 100000 + EPSILON ) val -= 100000;
+        else if ( val > 10000 + EPSILON ) val -= 10000;
+        else if ( val > 1000 + EPSILON ) val -= 1000;
+        else if ( val > 100 + EPSILON ) val -= 100;
+        else if ( val > 10 + EPSILON ) val -= 10;
+        else if ( val > 1 + EPSILON ) val -= 1;
+        else if ( val > 0.1 + EPSILON ) val -= 0.1;
+        else if ( val > 0.01 + EPSILON ) val -= 0.01;
+        else if ( val > 0.001 + EPSILON ) val -= 0.001;
+        else if ( val > 0.0001 + EPSILON ) val -= 0.0001;
+        else val = 0.0001;
+        return val;
+    }
+  
+    public boolean processKeyPressedEvent(KeyEvent keyEvent) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 127 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 245-252 into a separate method.